Codykilpatrick/Celestia
# feat(database): setup local database

Allows the creation of a local database.
**In order to use it you must have Postgres installed and running in the background.**
So far the only table that has a migration is the item table, we are going to need to add later on a migration for each of our other tables as well as migrations that seed the database.

We are using postgraphile for our API and db-migrate to handle the migrations.
**Test plan (required)**
1. Have Postgres installed and running in the background:
2. open a terminal and run the command `createdb celestia`
3. `cd packages/celestia-database`
4. run `npx db-migrate up` to get the migrations in.
5. `yarn db:connect` to launch the graphql interface
6. navigate to: `http://[::1]:5678/graphql` To view the graphQL gui.

Once thats open you should be able to run the following Queries/Mutations in the GUI.

To query the items table:
`
query itemsQuery {
  allItems {
    nodes {
      rowId
      id
      itemName
      typeId
    }
  }
}
`
To insert into the items table:
`
mutation CreateItem {
  createItem(input: { item: { typeId: 487, itemName: "Rifter" } }) {
    item {
      id
      typeId
      itemName
    }
  }
}
`
Then re-run the ItemsQuery and view your new stuff!

Let me know if you guys have any issues getting this setup.

Also the first ~30 minutes of this video is awesome for explaining everything and its how I got setup so I definitely recommend watching.
https://www.youtube.com/watch?v=eDZO8z1qw3k&t=2110s&ab_channel=knowthen
